src/parsita/state.py

```diff
@@ -126,20 +127,21 @@ def first(self) -> str:
         return self.source[self.position]
 
     @property
-    def rest(self) -> 'StringReader[str]':
+    def rest(self) -> 'StringReader':
         return StringReader(self.source, self.position + 1)
 
     @property
     def finished(self) -> bool:
         return self.position >= len(self.source)
 
-    def drop(self, count: int) -> 'StringReader[Input]':
+    def drop(self, count: int) -> 'StringReader':
         return StringReader(self.source, self.position + count)
 
     next_token_regex = re.compile(r'[\(\)\[\]\{\}\"\']|\w+|[^\w\s\(\)\[\]\{\}\"\']+|\s+')
 
     def next_token(self) -> str:
         match = self.next_token_regex.match(self.source, self.position)
+        assert match is not None
         return self.source[match.start():match.end()]
```

Comment:
if match is None:
            return self.source[match.start]
        else:
            return self.source[match.start():match.end()]

Reply:
The default `next_token_regex` cannot fail (I don't think), but you are right that this should be handled more elegantly. How about just returning the next character if the regex fails? This is only used for error messages.

Reply:
This change looks better than my `assert`, yeah.

Reply:
My original suggestion had a bug in it, so I fixed it and added a test. If you are satisfied with this PR, I'll merge it.